pymongo/mongo_client.py

```diff
@@ -624,7 +624,7 @@ def __init__(
         username = None
         password = None
         dbase = None
-        opts = {}
+        opts = common._CaseInsensitiveDictionary()
```

Comment:
Is this needed? We already seem to use _CaseInsensitiveDictionary for `keyword_opts` down on line 663.